service/src/main/scala/za/co/absa/loginsvc/rest/service/JWTService.scala

```diff
@@ -16,35 +16,61 @@
 
 package za.co.absa.loginsvc.rest.service
 
+import com.fasterxml.jackson.databind.{JsonNode, ObjectMapper}
 import com.nimbusds.jose.JWSAlgorithm
 import com.nimbusds.jose.jwk.{JWKSet, KeyUse, RSAKey}
 import io.jsonwebtoken.security.Keys
 import io.jsonwebtoken.{JwtBuilder, Jwts, SignatureAlgorithm}
+import org.slf4j.LoggerFactory
 import org.springframework.beans.factory.annotation.Autowired
 import org.springframework.stereotype.Service
+import software.amazon.awssdk.auth.credentials.DefaultCredentialsProvider
+import software.amazon.awssdk.regions.Region
+import software.amazon.awssdk.services.secretsmanager.SecretsManagerClient
+import software.amazon.awssdk.services.secretsmanager.model.GetSecretValueRequest
+import software.amazon.awssdk.services.secretsmanager.model.GetSecretValueResponse
 import za.co.absa.loginsvc.model.User
+import za.co.absa.loginsvc.rest.config.FetchSecretConfig
 import za.co.absa.loginsvc.rest.config.provider.JwtConfigProvider
 import za.co.absa.loginsvc.rest.service.JWTService.JwtBuilderExt
 import za.co.absa.loginsvc.utils.OptionExt
 
 import java.security.interfaces.RSAPublicKey
-import java.security.{KeyPair, PublicKey}
+import java.security.spec.{PKCS8EncodedKeySpec, X509EncodedKeySpec}
+import java.security.{KeyFactory, KeyPair, PublicKey}
 import java.time.Instant
 import java.time.temporal.ChronoUnit
-import java.util.Date
+import java.util.concurrent.{Executors, TimeUnit}
+import java.util.{Base64, Date}
 
 @Service
 class JWTService @Autowired()(jwtConfigProvider: JwtConfigProvider) {
 
+  private val logger = LoggerFactory.getLogger(classOf[JWTService])
+
   private val jwtConfig = jwtConfigProvider.getJWTConfig
-  private val rsaKeyPair: KeyPair = Keys.keyPairFor(SignatureAlgorithm.valueOf(jwtConfig.algName))
+  @volatile private var config: keyConfig = if(jwtConfig.fetchFromAws.nonEmpty)
+    {
+      logger.info("Fetching Keys from AWS Secrets Manager")
+      val awsConfig = jwtConfig.fetchFromAws.get
+      refreshSecrets(awsConfig)
+      fetchSecrets(awsConfig)
+    }
+  else
+    {
+      logger.info("Generating Keys in Memory")
+      val generateConfig = jwtConfig.generateInMemory.get
+      keyConfig(Keys.keyPairFor(SignatureAlgorithm.valueOf(generateConfig.algName)),
+        generateConfig.expTime,
+        generateConfig.algName)
+    }
```

Comment:
How about moving all this and with `fetchSecrets` and `refreshSecrets` to a different structure?
For example:
```scala 
trait KeyConfig {
  def algName: String
  def expTime: Int
  def keyPair: KeyPair
}

case class InMemoryKeyConfig(...) extends KeyConfig {...}
case class AwsKeyConfig(...) extends KeyConfig {
  def fetchSecrets() = ...
  ...
}

Currently, the `JwtService` class seems quite chaotic and contains multiple levels of detail.

Reply:
Implemented something along these lines